Remove maintenance from dev

hectormmg opened the pull request:
This PR:

- Removes `msal-core` and `maintenance` directories (they'll stay in msal-lts)
- Updates all `lib/msal-core` references to `tree/msal-lts/lib/msal-core`
- Removes `msal-core` from lerna scope
- Removes CI checks related to `msal-core`  (they will stay in the LTS branch)

hectormmg committed:
Update package-locks

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Move core to maintenance directory

hectormmg committed:
Update msal-core path references

hectormmg committed:
Remove msal-core from lerna scope

tnorling commented:
> Removes MSAL v1 from Issue templates

Are we ready to do this? We still get occasional issues about this (and even still list ADAL...)

hectormmg committed:
Remove msal-core from npm audit

hectormmg committed:
Add core and v3 to issue templates

hectormmg commented:
> > Removes MSAL v1 from Issue templates
> 
> Are we ready to do this? We still get occasional issues about this (and even still list ADAL...)

Reverted, also updated the core path and added an option for MSAL Browser v3 alpha issues, will check to see if labeler neesd an update for that

hectormmg committed:
Add v3 alpha labels to template bot

tnorling commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -23,7 +23,8 @@ body:
     label: Core Library
-    options: 
+    options:
+      - "MSAL.js v3-alpha (@azure/msal-browser@alpha)" 
       - "MSAL.js v2 (@azure/msal-browser)"
       - "MSAL.js v1 (@azure/msal or msal)"
```
Can we just do this instead? There's a version field below where people can specify

```suggestion
      - "MSAL.js (@azure/msal-browser)"
      - "MSAL.js v1 (@azure/msal or msal)"
```

hectormmg replied:
I think the only downside is we won't have different labeling for v3. I don't have a preference either way.

tnorling replied:
If there's a need to differentiate we can update the labeler to read the version field

tnorling replied:
I just don't want to update these templates every time we push a new major version :)

tnorling commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -11,7 +11,7 @@ body:
       - [@azure/msal-angular](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/FAQ.md)
       - [@azure/msal-react](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/FAQ.md)
-      - [@azure/msal or msal](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/docs/FAQ.md)
+      - [@azure/msal or msal](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/maintenance/msal-core/docs/FAQ.md)
```
Please update the other templates as well

hectormmg replied:
The others don't have a list like this.

tnorling replied:
The other templates have the library selector though

tnorling commented on lib/msal-browser/README.md:
```diff
@@ -29,9 +29,9 @@
+The `@azure/msal-browser` package described by the code in this folder uses the [`@azure/msal-common` package](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-common) as a dependency to enable authentication in JavaScript Single-Page Applications without backend servers. This version of the library uses the OAuth 2.0 Authorization Code Flow with PKCE. To read  ...
 
-This is an improvement upon the current `msal-core` library which will utilize the authorization code flow in the browser. Most features available in the old library will be available in this one, but there are nuances to the authentication flow in both. The `@azure/msal-browser` package does NOT support the implicit flow.
+This is an improvement upon the previous `msal-core` library which will utilize the authorization code flow in the browser. Most features available in the old library will be available in this one, but there are nuances to the authentication flow in both. The `@azure/msal-browser` package does NOT support the implicit flow.
```
```suggestion
This is an improvement upon the previous `@azure/msal` library which will utilize the authorization code flow in the browser. Most features available in the old library will be available in this one, but there are nuances to the authentication flow in both. The `@azure/msal-browser` package does NOT support the implicit flow.
```

tnorling commented on package.json:
```diff
@@ -58,7 +58,6 @@
     "lint:angular": "eslint lib/msal-angular/src --ext .ts",
     "lint:browser": "eslint lib/msal-browser/src --ext .ts",
     "lint:common": "eslint lib/msal-common/src --ext .ts",
-    "lint:core": "eslint lib/msal-core/src --ext .ts",
```
Rather than removing, let's edit the path. This step will still be run in CI if we need to make changes to the lib

hectormmg replied:
I'd like for us to discuss the merit of having things in the "maintenance" directory now that we have an LTS branch. Should we be making changes to unsupported maintenance libs in the dev branch or on LTS? I would argue that these changes should be made in LTS because they are out of active development.

tnorling replied:
I have no strong opinion but I agree these should exist on one branch or the other, not both

tnorling commented on release-scripts/create-releases.js:
```diff
@@ -32,7 +32,6 @@ const defaultLibFolders = [
     "lib/msal-angular",
     "lib/msal-browser",
     "lib/msal-common",
-    "lib/msal-core",
```
Rather than removing, let's edit the path

tnorling commented on typedoc.json:
```diff
@@ -5,11 +5,10 @@
         "lib/msal-node/src/index.ts",
         "lib/msal-react/src/index.ts",
         "lib/msal-angular/src/public-api.ts",
-        "lib/msal-core/src/index.ts"
```
Same here

sameerag commented on .github/workflows/npm-audit.yml:
```diff
@@ -55,7 +55,6 @@ jobs:
         fail-fast: false
         matrix:
           library:
-            - msal-core
```
We still need to run this in maintenance folder right? npm-audit identifies security bugs?

hectormmg replied:
It'll still be in the msal-lts branch. These audits run on PRs so we will still only see it when changes are made to core on the LTS branch.

sameerag commented on lib/msal-browser/README.md:
```diff
@@ -29,9 +29,9 @@
+The `@azure/msal-browser` package described by the code in this folder uses the [`@azure/msal-common` package](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-common) as a dependency to enable authentication in JavaScript Single-Page Applications without backend servers. This version of the library uses the OAuth 2.0 Authorization Code Flow with PKCE. To read  ...
 
-This is an improvement upon the current `msal-core` library which will utilize the authorization code flow in the browser. Most features available in the old library will be available in this one, but there are nuances to the authentication flow in both. The `@azure/msal-browser` package does NOT support the implicit flow.
+This is an improvement upon the previous `msal-core` library which will utilize the authorization code flow in the browser. Most features available in the old library will be available in this one, but there are nuances to the authentication flow in both. The `@azure/msal-browser` package does NOT support the implicit flow.
```
This entire statement feels a little round about. Can we be more succinct and say - "This is an improved auth experience that retains all important features from `@azure/msal` with enhanced security and ease of use. This package DOES NOT support the implicit flow and adapts auth-code flow as mentioned above."

hectormmg replied:
I'd rather not make any unnecessary changes here. The README is likely to change when v3 is GA anyway, core probably won't even be mentioned. This change is just for the sake of keeping it correct.

hectormmg committed:
Update lib/msal-browser/README.md

Co-authored-by: Thomas Norling <thomas.l.norling@gmail.com>

hectormmg committed:
Merge branch 'dev' into core-maintenance

hectormmg committed:
Update .github/ISSUE_TEMPLATE/bug_report.yml

Co-authored-by: Thomas Norling <thomas.l.norling@gmail.com>

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' of github.com:AzureAD/microsoft-authentication-library-for-js into dev

hectormmg committed:
Merge branch 'dev' into core-maintenance

hectormmg committed:
Remove maintenance directory from dev branch

hectormmg committed:
Update mentions of msal-core to point to lts branch

hectormmg committed:
Merge branch 'dev' into core-maintenance

hectormmg committed:
Update issue templates to remove v2 form MSAL Browser dropdown

hectormmg committed:
Merge branch 'core-maintenance' of github.com:AzureAD/microsoft-authentication-library-for-js into core-maintenance

tnorling commented on .github/ISSUE_TEMPLATE/question.yml:
```diff
@@ -23,7 +23,8 @@ body:
     label: Core Library
-    options: 
+    options:
+      - "MSAL.js v3-alpha (@azure/msal-browser@alpha)" 
       - "MSAL.js v2 (@azure/msal-browser)"
```
```suggestion
      - "MSAL.js (@azure/msal-browser)"
```

tnorling replied:
Please also make this change in all templates

hectormmg replied:
I thought I did, sorry about that.

hectormmg replied:
Didn't push the changes apparently, here they go...

tnorling commented on .github/issue_template_bot.json:
```diff
@@ -3,58 +3,91 @@
+                    "searchStrings": [
+                        "@azure/msal-browser"
+                    ]
+                },
+                "msal-browser-v3": {
+                    "searchStrings": [
+                        "@azure/msal-browser@alpha"
+                    ]
```
```suggestion
```

tnorling commented on README.md:
```diff
@@ -20,21 +20,21 @@ The [`lib`](https://github.com/AzureAD/microsoft-authentication-library-for-js/t
 - [Microsoft Authentication Library for Angular v2.x](lib/msal-angular/): A wrapper of the msal-browser 2.x library for apps using Angular framework.
 
-- [Microsoft Authentication Library for JavaScript v1.x](lib/msal-core/): A browser-based, framework-agnostic core library that enables authentication and token acquisition with the Microsoft Identity platform in JavaScript applications. Implements the OAuth 2.0 [Implicit Grant Flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow), and is [OpenID-compliant ...
+- [Microsoft Authentication Library for JavaScript v1.x](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/msal-lts/lib/msal-core): A browser-based, framework-agnostic core library that enables authentication and token acquisition with the Microsoft Identity platform in JavaScript applications. Implements the OAuth 2.0 [Implicit Grant Flow](https://docs.microsoft.com/azure/a ...
```
You want to move this down to the "maintenance" section?

hectormmg committed:
Update .github/issue_template_bot.json

Co-authored-by: Thomas Norling <thomas.l.norling@gmail.com>

hectormmg committed:
Merge branch 'core-maintenance' of github.com:AzureAD/microsoft-authentication-library-for-js into core-maintenance

hectormmg committed:
Move reference to MSAL v1 to maintenance section in README

jo-arroyo commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -11,7 +11,7 @@ body:
       - [@azure/msal-angular](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/FAQ.md)
       - [@azure/msal-react](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/FAQ.md)
-      - [@azure/msal or msal](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/docs/FAQ.md)
+      - [@azure/msal or msal](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/msal-lts/lib/msal-core/docs/FAQ.md)
```
Question: If we're referring to browser as `MSAL.js` below, and core as `MSAL.js v1`, would we not want to specify this as msal v1 as well here?

hectormmg replied:
I don't have a preference. I think this is more just matching the package name.

hectormmg committed:
Merge branch 'dev' into core-maintenance

hectormmg committed:
Merge branch 'dev' into core-maintenance

hectormmg committed:
Merge branch 'dev' into core-maintenance

hectormmg committed:
Replace credscan suppression file

hectormmg committed:
Remove CredScanSuppression file and sdl-task reference to it

hectormmg committed:
Merge branch 'dev' into core-maintenance

tnorling commented on lib/msal-angular/docs/v1-docs/v0-v1-upgrade-guide.md:
```diff
@@ -22,7 +22,7 @@ Steps:
 
 Steps:
-1. The first argument is the configuration object, which is the [same `Configuration` object](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/src/Configuration.ts) you would pass to `msal`.
+1. The first argument is the configuration object, which is the [same `Configuration` object](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/msal-lts/dev/msal-core/src/Configuration.ts) you would pass to `msal`.
```
Should this be msal-lts/lib ?

hectormmg replied:
It in fact should, thanks for the catch

hectormmg committed:
Fix incorrect link in v0-v1 migration guide

hectormmg committed:
Merge branch 'core-maintenance' of github.com:AzureAD/microsoft-authentication-library-for-js into core-maintenance

sameerag reviewed (approved):
Nit: Instead of `msal v1` in some of the docs, we can just do `msal-core` as we are not versioning after that. Approved.

hectormmg committed:
Merge branch 'dev' into core-maintenance
